rsocket/RSocketClient.h

```diff
@@ -4,66 +4,83 @@
 #include "rsocket/RSocketNetworkStats.h"
 #include "rsocket/RSocketParameters.h"
+#include "rsocket/RSocketRequester.h"
 #include "rsocket/RSocketStats.h"
+#include "rsocket/ResumeManager.h"
 
 namespace rsocket {
 
+class RSocket;
 class RSocketConnectionManager;
 
 /**
- * API for connecting to an RSocket server. Returned from RSocket::createClient.
- *
+ * API for connecting to an RSocket server. Created with RSocket class.
  * This connects using a transport from the provided ConnectionFactory.
  */
 class RSocketClient {
  public:
-  explicit RSocketClient(std::unique_ptr<ConnectionFactory>);
-  ~RSocketClient(); // implementing for logging right now
+  ~RSocketClient();
 
   RSocketClient(const RSocketClient&) = delete; // copy
-  RSocketClient(RSocketClient&&) = delete; // move
+  RSocketClient(RSocketClient&&) = default; // move
   RSocketClient& operator=(const RSocketClient&) = delete; // copy
-  RSocketClient& operator=(RSocketClient&&) = delete; // move
-
-  /*
-   * Connect asynchronously and return a Future which will deliver the RSocket
-   *
-   * Each time this is called:
-   * - a new connection is created using a ConnectionFactory
-   *
-   * The returned RSocketRequester is retained by the RSocketClient instance
-   * so that it lives for the life of RSocketClient, as opposed to the scope
-   * of the Future (such as when used with Future.then(...)).
-   *
-   * To destruct a single RSocketRequester sooner than the RSocketClient
-   * call RSocketRequester.close().
-   */
-  folly::Future<std::unique_ptr<RSocketRequester>> connect(
-      SetupParameters setupParameters = SetupParameters(),
-      std::shared_ptr<RSocketResponder> responder = std::shared_ptr<RSocketResponder>(),
-      std::unique_ptr<KeepaliveTimer> keepaliveTimer = std::unique_ptr<KeepaliveTimer>(),
-      std::shared_ptr<RSocketStats> stats = std::shared_ptr<RSocketStats>(),
-      std::shared_ptr<RSocketNetworkStats> networkStats = std::shared_ptr<RSocketNetworkStats>());
-
-  // TODO implement version supporting fast start (send SETUP and requests
-  // without waiting for transport to connect and ack)
-  //  std::shared_ptr<RSocketRequester> fastConnect();
-
-  std::unique_ptr<RSocketRequester> fromConnection(
-      std::unique_ptr<DuplexConnection> connection,
-      folly::EventBase& eventBase,
-      SetupParameters setupParameters = SetupParameters(),
-      std::shared_ptr<RSocketResponder> responder = std::shared_ptr<RSocketResponder>(),
-      std::unique_ptr<KeepaliveTimer> keepaliveTimer = std::unique_ptr<KeepaliveTimer>(),
-      std::shared_ptr<RSocketStats> stats = std::shared_ptr<RSocketStats>(),
-      std::shared_ptr<RSocketNetworkStats> networkStats = std::shared_ptr<RSocketNetworkStats>());
+  RSocketClient& operator=(RSocketClient&&) = default; // move
+
+  friend class RSocket;
+
+  // Returns the RSocketRequester associated with the RSocketClient.
+  std::shared_ptr<RSocketRequester> getRequester() const;
```

Comment:
The plan is to make it such that, if the application has either of Requester or RSocketClient, it should be fine.  The code to make the Requester sufficient to keep the RSocket state is not in yet.